akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.typed.scaladsl.Behaviors
+
+object SpawnProtocol {
```

Comment:
I thought we want to call it not SpawnProtocol since this limits any future expansion in case we'd want to?
RootProtocol / UserGuardianProtocol / ActorSystemProtocol / TopLevelProtocol?

Reply:
It isn't necessarily the top level actor/guardian so I think this is better.

Reply:
What isn't? This actor certainly is. If you mean the protocol be used by others then a protocol type hierarchy can be created.

Reply:
It can be used at other places than the root in the actor hierarchy.

AnyKindOfFutureIdea? - nah

Reply:
> It can be used at other places than the root in the actor hierarchy.
> AnyKindOfFutureIdea? - nah

You are completely turning around what I'm saying here...

Is it THE protocol of THE ActorSystem? I do find this weird and limiting.
Is it A protocol that could be used in other places, yeah I guess so; this was not really the goal of this excercise though and I wonder if it would not be then used in wrapped messages in any case -- `RouterCreateThing(Spawn())` rather than raw, so I feel we're jumping the shark here.

Reply:
it's not THE protocol, it's still something you opt in to when starting the system

Reply:
Hm.. I see; this is because we're not doing the "our implementation" of it for making `val system: ActorRef[...] = ActorSystem()`  it is only opt in for something users could implement... I see, sorry it took me a moment to spot this.

In this shape and the next ticket I think we're okey here. Thanks for bearing with me

Reply:
I'm also interpreting you like that.

It the behavior/actor is useful also in other places than as guardian. For example, you have a HTTP endpoint part of your system that spawns actors for some incoming requests, spawning those isn't necessarily good as top-level but lower down in the actor tree.

Reply:
Refresh ;)

Reply:
I'm too slow for this conversation :D